Chrysalis Docs Review IV (#72)
iotaledger/chrysalis-docs: docs/introduction/path_to_chrysalis.md

The benefit of this approach is that many of the protocol properties will remain the same, or almost the same, for Coordicide. That, together with a better set of developer tools, will make the transition to Coordicide much simpler.

> 

Reviewer: **For line 5** I'd add a comma after 2016
```suggestion 
The IOTA mainnet has been operational since 2016, and the overall engineering strategy evolved significantly based on industry demand and feedback.
```

The intended outcomes for Chrysalis were:

- **A simpler transition to Coordicide** - With Coordicide making significant progress, we wanted to ensure that all developers and companies that built and deployed on Chrysalis would have as smooth a transition to Coordicide as possible.
- **Substantial performance improvements** - With the changes introduced by Chrysalis, we saw a substantial improvement in the scalability and reliability of the IOTA Mainnet.

> - **Improved developer and user experience** - The new protocol features, new libraries, and the new wallet madeIOTA one of the best platforms to build on, removing the friction which developers experience today while allowing solutions built on top of the protocol to provide better user experience.

Reviewer: ```suggestion
**Improved developer and user experience** - The new protocol features, new libraries, and the new wallet made IOTA one of the best platforms to build on, removing the friction which developers experience today while allowing solutions built on top of the protocol to provide better user experience.
```

As well, this sentence feels a bit long, but I'm not sure how I'd split it.

> - **Accelerated adoption** - Chrysalis made IOTA production-ready by becoming a stable protocol, with a reliable set of developer tools and frameworks that will enabled startups, corporations, and governments to develop and launch products powered by IOTA.

Reviewer: ```suggestion
- **Accelerated adoption** - Chrysalis made IOTA production-ready by becoming a stable protocol, with a reliable set of developer tools and frameworks that will enable startups, corporations, and governments to develop and launch products powered by IOTA
```

## The Stages to Chrysalis

> ![](/Users/charlesthompson/chrysalis-docs/docs/introduction/assets/path_to_chrysalis/01.png)

Reviewer: please correct this route, and add an alt text

The Chrysalis upgrade was a complex undertaking. We coordinated a number of distinct products to ensure a smooth transition for IOTA’s current users and partners. In addition to the core node software, we also needed to update our wallet software, our libraries, and the entire infrastructure.  

Another important requirement was the ease of transition to the future Coordicide network. By carefully planning the breaking changes introduced along the way, and providing support in our developer tools, we ensured that our growing ecosystem of developers, startups, and corporations could reliably develop and launch new innovative products on IOTA.  

The plan for implementing Chrysalis was divided into two phases.

> **The first phase** consisted of improved tip selection (URTS), milestone selection, and White flag. These were implemented in the node software gradually. This phase required an upgrade of all nodes, including the coordinator node, and did not require a snapshot.  

Reviewer: Please clarify that URTS stand for Uniform Random Tip Selection, and add a link to https://github.com/luca-moser/protocol-rfcs/blob/rfc-urts-tip-sel/text/0008-weighted-uniform-random-tip-selection/0008-weighted-uniform-random-tip-selection.md

The first phase of Chrysalis resulted in:

- Transaction confirmation times of around 10 seconds.
- Transactions rarely needing reattachment.
- A substantial TPS increase.
- Performance and reliability improvements for nodes.

> **The second phase** of Chrysalis consisted of adopting and/or implementing UTXO, atomic transactions, reusable addresses (Ed25519), a transition to a binary transaction layout, and a new set of client libraries and developer tools. These represented significant changes to the core protocol and the way transactions were structured. Once everything was tested, validated, and audited, the Foundation deployed a new Chrysalis network. The upgrade consisted of an extended period when the legacy network remained operational which allowed users, exchanges, and partners to migrate to the Chrysalis network at will. 

Reviewer: - Did we adopt, implement or both ? 
- We could clarify that UTXO stands for Unspent transaction output, and add a link to https://github.com/iotaledger/stronghold.rs/tree/dev/.changes
- We could add a link to the Ed25519, reusable addreses and such when we mention them. 
- "These represented", is these the new set of client libraries and dev tools? Or all the upgrades in general?

> The second phase of Chrysalis consisted of:

Reviewer: This seems a bit redundant, as the previous paragraph has the exact same opening, and raises the more or less same points.
Perhaps we could reshape this list and the following paragraph into what were the results of the second phase.  Such as improving usability and such.

- Reusable addresses and support for more standard cryptography (EdDSA), making efficient hardware support for all major architectures possible.
- A simplified transaction layout and a reduction in transaction size, further increasing performance and efficiency.
- Significant improvements to the usability and reliability of IOTA.
- A switch to a UTXO based model from the previous account model.

The introduction of reusable addresses was an important change for token holders. This vastly improved IOTA’s usability and made integration into new exchanges, wallets, and payment systems much simpler. A new wallet, called Firefly, was released with Chrysalis which allowed token holders to transition from the WOTS address scheme to the new EdDSA scheme.  

### From Plan to Action

With Chrysalis, we had to make decisions on how to best implement all the upcoming changes, ensuring a correct and timely implementation. What follows is an overview of the various components we worked on to successfully implement this update.

### Specification and Standardization

> Specifications are a major part of our new development process. All our new software projects (Node software, Wallet, Identity, Access, Streams, etc.) were based on vetted specifications. Specifications make it possible for external parties to follow a project’s intended functionality, for example for audit purposes, or develop their own implementations in different languages.

Reviewer: - we should add links to the mentioned projects.
- I don't think we should etc. our projects.

### Wallet Support

The Trinity wallet is a popular IOTA wallet. With Chrysalis, we released a new wallet implementation, [Firefly](https://blog.iota.org/firefly-iota-next-generation-wallet-26bdd4d01510/). The team worked on a complete reevaluation of the wallet architecture, with a completely redesigned user experience. At its core is a [new wallet library](../libraries/wallet.md) written in Rust. The wallet library is designed to allow other developers to easily implement IOTA wallets within their applications. Another new Rust library utilized by Firefly is Stronghold; Stronghold enables ultra-secure secret handling and storage.  

### Infrastructure

The IOTA Foundation previously supported two public Networks: Mainnet and Devnet (Devnet is for PoCs and application tests). Both networks provided public endpoints for users and partners. For reference, you can read more about the networks and how you can participate in our [documentation](https://docs.iota.org/docs/getting-started/1.1/networks/overview).  

> Mainnet was replaced by the Chrysalis network where Devnet was simply upgraded after the Chrysalis release.

Reviewer: ```suggestion
Mainnet was replaced by the Chrysalis network, while Devnet was simply upgraded after the Chrysalis release.
```

> 

Reviewer: **For line 91**
```suggestion
In case you need single node endpoints that expose native MQTT, you can use the following urls:
```
